opencog/query/InitiateSearchCB.cc

```diff
@@ -761,7 +761,7 @@ bool InitiateSearchCB::variable_search(PatternMatchEngine *pme)
 			if (prev != _pattern) { prev = _pattern; count = 0; }
 			else {
 				count++;
-				if (5 < count)
+				if (100 < count)
 					throw RuntimeException(TRACE_INFO,
```

Comment:
I don't think hitting more than 1x is a bug. We're talking about the invariance of _pattern only, I don't think that is the only thing to look at. If it was really looping ad infinitum, 2, 5 or 100 would amount to the same result.